### PR TITLE
Exposing deployment strategy + fix ORY Hydra resources

### PR DESCRIPTION
**Pull request by lumi017**
**Description**

Changes proposed in this pull request:

- Exposing deployment strategy in dex, ory and compass helm charts
- Fix ORY Hydra resources

**Related issue(s)**

### CODE REVIEW

File: resources/ory/values.yaml [hunk 1]
```diff
@@ -80,13 +80,14 @@ hydra:
...
     dangerousAllowInsecureRedirectUrls: false
     existingSecret: "ory-hydra-credentials"
-  resources:
-    limits:
-      cpu: 100m
-      memory: 256Mi
-    requests:
-      cpu: 50m
-      memory: 128Mi
+    deployment:
```
Comment (ebensom) [inline resources/ory/values.yaml:83]: resources: are now under hydra.hydra., it should be moved one level upper to hydra.

File: resources/compass/charts/cockpit/values.yaml [hunk 2]
```diff
@@ -19,3 +19,9 @@ service:
 deployment:
   image:
     pullPolicy: IfNotPresent
+  # By default deploymentStrategy is set to rollingUpdate with maxSurge of 25% and maxUnavailable of 25% . you can change type to `Recreate` or can uncomment `rollingUpdate` specification and adjust them to your usage.
+  strategy: {}
+    # rollingUpdate:
+    #   maxSurge: 25%
+    #   maxUnavailable: 25%
+    # type: RollingUpdate
```
Comment (pkosiec) [inline resources/compass/charts/cockpit/values.yaml:27]: Hmm I wonder if we need these two comments here.
I would suggest something more condensed, like:

```yaml
strategy: {} # Read more: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
```

If you like this idea, please change it in every chart.

File: resources/ory/values.yaml (hunk 1 above)
Comment (lumi017) [reply resources/ory/values.yaml]: ooups, fixed

File: resources/compass/charts/cockpit/values.yaml (hunk 2 above)
Comment (lumi017) [reply resources/compass/charts/cockpit/values.yaml]: sure, changed

File: resources/compass/charts/director/values.yaml [hunk 3]
```diff
@@ -10,6 +10,14 @@ deployment:
   dbPool:
     maxOpenConnections: 30
     maxIdleConnections: 2
+  liveness:
```
Comment (ebensom) [inline resources/compass/charts/director/values.yaml:13]: Please remove liveness: and readiness:, I think those should have been added by a faulty merge from upstream.

Comment (lumi017) [reply resources/compass/charts/director/values.yaml]: cleaned up

**Pull request merged**